### ISSUE TITLE
Cannot make a build due to UnityEditor namespace in plugin

### ISSUE THREAD

**Issue by kyleshepherd** [MagmarFire/Enso-Music-Player#28]
I tried to make a build of my game which uses this plug in, and I receive an error saying that "The type or namespace name 'UnityEditor' could not be found", originating from `MusicTrack.cs`

If I move the `MusicTrack` script to an Editor folder, I then get multiple errors stating "The type or namespace 'MusicTrack' could not be found"

It seems as though a build cannot be created whilst using this plugin currently. If anyone knows a fix, please advise

**Comment by MagmarFire** [comment]
I have committed an emergency fix for this issue. Before, I was trying to get the file path for audio clip assets in scripting so developers don't need to do it themselves. However, that approach doesn't work on build, so for now, developers will be required to include the path to the audio clip if they want the player to set loop points automatically from metadata.

This causes an API change, which I will document at a later time. If you're setting the loop points directly in the Loop Points section for each song, you shouldn't need to do anything. However, if you are doing it through metadata (or if you're allowing the system to compensate for any frequency changes), you'll need to do one more thing before building. In the Loop Points section for each track, there's now a field called Relative File Path. Enter path for the file relative to the Assets folder into it. For example, if you have an audio clip called `Test.ogg` located in the Music folder, you'd type in `Music/Test.ogg`.

It's really rudimentary and totally contrary to how I want the system to be designed--that is, not having to make developers enter data--but as this is an emergency fix, we'll make do. Feel free to get the latest version to test these changes. If you're able to build, I'll close this ticket and start work on making the process more streamlined and simpler.

In the meantime, sorry for the extra added step.

**Comment by MagmarFire** [comment]
I've released v1.4.1 to address the issue, which solely incorporates the emergency fix from before. Thank you again for bringing this to my attention.

After doing more research into how I can eliminate the need to specify relative file paths, I've concluded that it's a non-trivial endeavor. Unity, unfortunately, does not offer any particular straightforward, build-supported method to getting something as simple as a file path for an asset during runtime--which is needed to read the original file's metadata. As such, specifying them manually will likely not be made redundant until Unity implements a utility function that can be used in builds.

I'll still add it as a low-priority issue, but it won't be completed anytime soon. Until then, I'll update the API reference to take into account this update.

Again, though, if you're not using file metadata at all, and if you're using a standard frequency of, say, 32kHz, then you shouldn't need to do anything extra if you apply this patch. Just keep this in mind.

As the release should address the issue, I'm closing the ticket. If you notice that it's still not building, feel free to open it back up.
